### PR TITLE
Log the driver version + plat info on connect to aid in log analysis

### PR DESCRIPTION
**Pull request by David-Engel**
To aid in log analysis. I was tempted to log all system properties, but there are some sensitive ones in there, so I targeted the logging to relevant properties.

This is not a requirement to merge. I'm open to discussion. It just seems like this might help when we get issues where people leave out important details.

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 1]
```diff
@@ -903,11 +903,30 @@ static String getPropertyOnlyName(String name, Logger logger) {
...
         loggerExternal.entering(getClassNameLogging(), "connect", "Arguments not traced.");
         SQLServerConnection result = null;
 
+        if (loggerExternal.isLoggable(Level.FINE)) {
+            loggerExternal.log(Level.FINE, "Microsoft JDBC Driver " + SQLJdbcVersion.major + "." + SQLJdbcVersion.minor +
+                    "." + SQLJdbcVersion.patch + "." + SQLJdbcVersion.build + SQLJdbcVersion.releaseExt + " for SQL Server");
+            if (loggerExternal.isLoggable(Level.FINER)) {
+                for (String propertyKeyName : systemPropertiesToLog) {
+                    String propertyValue = System.getProperty(propertyKeyName);
+                    if (propertyValue != null && !propertyValue.isEmpty()) {
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:922]:
```suggestion
                    if (null != propertyValue && !propertyValue.isEmpty()) {
```

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 2]
```diff
@@ -903,11 +903,30 @@ static String getPropertyOnlyName(String name, Logger logger) {
         return null;
     }
 
+    private final static String[] systemPropertiesToLog = new String[] { "java.specification.vendor",
+            "java.specification.version", "java.class.path", "java.class.version", "java.runtime.name",
+            "java.runtime.version", "java.vendor", "java.version", "java.vm.name", "java.vm.vendor", "java.vm.version",
+            "java.vm.specification.vendor", "java.vm.specification.version", "os.name", "os.version", "os.arch" };
```
Comment (cheenamalhotra) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:909]: I'm not very sure if these System properties are "required" to be logged by the driver. Reason being, they are not owned/provided by the driver, if anyone wants to log, they could do themselves as they're readily available in JVM.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java (hunk 1 above)
Comment (David-Engel) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:922]: That style honestly drives me nuts. It just reads unnaturally to me. I've read the arguments for both styles, and there is no real advantage to one over the other in Java, that I'm aware of.

Our [Coding Guidelines](https://github.com/microsoft/mssql-jdbc/blob/dev/Coding_Guidelines.md) don't mention it, either. Is this an unspoken style the team follows?

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java (hunk 2 above)
Comment (David-Engel) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java:909]: The intent is to gather information that is sometimes left out during support but that affects driver functionality. I've encountered issues that can be traced to differences around these settings.

**Review by dibianalvarez — APPROVED** [review]

**Review by dibianalvarez — APPROVED** [review]

**Review by v-zhangw — APPROVED** [review]

**Review by lilgreenbird — APPROVED** [review]

**Pull request merged**